src/Repository/AbstractDatabaseMessageRepository.php

```diff
@@ -333,6 +297,41 @@ protected function setClassMessageFactory($classFactory)
     }
 
     /**
+     * Run a query
+     *
+     * @param string $query
+     * @return \Doctrine\DBAL\Driver\Statement|\PDOStatement
+     * @throws DBALException
+     */
+    private function executeQuery($query)
```

Comment:
use specific try / catch in the right child class (for PDO & DBAL) with the good catch Exception.
And check the exception code instead of the exception message (or both with ||)